query-dsl/src/main/java/org/infinispan/query/dsl/continuous/ContinuousQuery.java

```diff
@@ -0,0 +1,23 @@
+package org.infinispan.query.dsl.continuous;
+
+import org.infinispan.query.dsl.Query;
+
+import java.util.List;
+
+/**
+ * A container of continuous query listeners for a cache.
+ * <p>Implementations are not expected to be threadsafe.
+ *
+ * @author anistor@redhat.com
+ * @since 8.2
+ */
+public interface ContinuousQuery<K, V> {
+
+   <C> void addContinuousQueryListener(Query query, ContinuousQueryListener<K, C> listener);
+
+   void removeContinuousQueryListener(ContinuousQueryListener<K, ?> listener);
+
+   List<ContinuousQueryListener<K, ?>> getListeners();
+
+   void removeAllListeners();
+}
```

Comment:
I moved the CQ interfaces to ```org.infinispan.query.api.continuous``` package in query-dsl module. This required some hacking of the osgi imports/exports. 

I'd also rename query-dsl module to query-api in future.